api.go

```diff
@@ -528,32 +521,29 @@ func (lapi *Client) DeleteStream(id string) error {
 
 // CreateStreamEx creates stream with specified name and profiles
 func (lapi *Client) CreateStreamEx(name string, record bool, presets []string, profiles ...Profile) (*CreateStreamResp, error) {
-	return lapi.CreateStreamEx2(name, record, "", presets, profiles...)
+	return lapi.CreateStream(CreateStreamReq{Name: name, Record: record, Presets: presets, Profiles: profiles})
 }
 
-// CreateStreamEx creates stream with specified name and profiles
-func (lapi *Client) CreateStreamEx2(name string, record bool, parentID string, presets []string, profiles ...Profile) (*CreateStreamResp, error) {
-	// presets := profiles
-	// if len(presets) == 0 {
-	// 	presets = lapi.presets
-	// }
-	glog.Infof("Creating Livepeer stream '%s' with presets '%v' and profiles %+v", name, presets, profiles)
-	reqs := &createStreamReq{
-		Name:    name,
-		Presets: presets,
-		Record:  record,
+// CreateStream creates stream with specified name and profiles
+func (lapi *Client) CreateStream(csr CreateStreamReq) (*CreateStreamResp, error) {
+	if csr.Name == "" {
+		return nil, errors.New("stream must have a name")
 	}
-	if len(presets) == 0 {
-		reqs.Profiles = StandardProfiles
+	if len(csr.Presets) == 0 && len(csr.Profiles) == 0 {
+		csr.Profiles = StandardProfiles
 	}
-	if len(profiles) > 0 {
-		reqs.Profiles = profiles
+	glog.Infof("Creating Livepeer stream '%s' with presets '%v' and profiles %+v", csr.Name, csr.Presets, csr.Profiles)
+	b, err := json.Marshal(csr)
+	if err != nil {
+		glog.V(logs.SHORT).Infof("Error marshalling create stream request %v", err)
+		return nil, err
 	}
```

Comment:
Should not have this logic here since `lapi.newRequest` already handles request payload JSON serialization